src/build.ts

```diff
@@ -0,0 +1,100 @@
+import { flavorEntries, version } from "@catppuccin/palette";
+import { fileIcons } from "./vscode-icons/src/defaults/fileIcons.ts";
+
+console.log("-- ctp/palette v" + version);
+
+const ZED_THEME = {
+  "$schema": "https://zed.dev/schema/icon_themes/v0.2.0.json",
+  "name": "Catppuccin Icons",
+  "author": "Catppuccin <releases@catppuccin.com>",
+  "themes": []
+};
+
+const THEME_OVERRIDES = {
+  "chevron_icons": {
+    "collapsed": null,
+    "expanded": null
+  },
+  "file_stems": {
+    "LICENSE": "license",
+    "README": "readme",
+    "Makefile": "makefile",
+    "Caddyfile": "caddy",
+    "Dockerfile": "docker"
+  }
+}
+
+// stage 1: create `file_suffixes` for zed schema
+// example: `{ "aep": "adobe-ae" }`
+let suffixEntries = {};
+suffixEntries["file_suffixes"] = {};
+for (const [vsCodeFileIcons_key, vscodeFileIcons_value] of Object.entries(fileIcons)) {
+
+  vscodeFileIcons_value.fileNames?.forEach(data => {
+    if (!suffixEntries.hasOwnProperty(data)) {
+      suffixEntries["file_suffixes"][data] = vsCodeFileIcons_key;
+    } else {
+      console.warn(`-- Key '${vsCodeFileIcons_key}' already exists, appending anyway...`);
+      suffixEntries["file_suffixes"][data] = vsCodeFileIcons_key;
+    }
+  });
+
+  vscodeFileIcons_value.fileExtensions?.forEach(data => {
+    if (!suffixEntries.hasOwnProperty(data)) {
+      suffixEntries["file_suffixes"][data] = vsCodeFileIcons_key;
+    } else {
+      console.warn(`-- Key '${vsCodeFileIcons_key}' already exists, appending anyway...`);
+      suffixEntries["file_suffixes"][data] = vsCodeFileIcons_key;
+    }
+  });
+}
+
+// stage 2: compile flavor specific properties
+let THEME = {};
+const fileIconEntries = {};
+fileIconEntries["file_icons"] = {};
+const flavorProperties: { [key: string]: any } = {};
+
+function createDirectoryIcons(flvr: string) {
+  return {
+    directory_icons: {
+      collapsed: `./icons/${flvr}/_folder.svg`,
+      expanded: `./icons/${flvr}/_folder_open.svg`
+    }
+  };
+}
+
+flavorEntries.map(([_, flavor]) => {
+
+  const flvr = flavor.name.replace('é', 'e').toLowerCase();
```

Comment:
You should be able to use the identifier instead of the name here iirc

Reply:
ohhhh, so something like this? I figured it'd be a property of `flavor`; doing `flavor.id` resulted in `undefined` lol

```typescript
flavorEntries.map(([id, flavor]) => {
...
const directoryIconsEntries = createDirectoryIcons(id);
```

I'm still a novice at the language 😅